### PR TITLE
move rx and rxjs to dependencies. Update remaining dependencies

### PR DESCRIPTION
**Pull request by kentmclean**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,18 +16,20 @@
...
-    "browserslist-config-canopy": "^2.0.0",
+    "babel-jest": "^24.7.1",
+    "browserslist-config-canopy": "^3.1.0",
     "in-publish": "^2.0.0",
-    "jest": "^23.6.0",
-    "rimraf": "^2.6.2",
+    "jest": "^24.7.1",
+    "rimraf": "^2.6.3"
+  },
+  "dependencies": {
```
Comment (kentmclean) [inline package.json]: shifting `rx` and `rxjs` to `dependencies` reflects the reality that these two libraries are in fact required at runtime. There are also several alternatives to this, including using SystemJS to check/load these dependencies at runtime. Another would be to add them in a `peerDependencies` block, though this option doesn't seem like a good one.

File: package.json [hunk 2]
```diff
@@ -1,8 +1,8 @@
 {
   "name": "disposable-component",
-  "version": "0.3.2",
+  "version": "0.4.0",
```
Comment (kentmclean) [inline package.json]: version bump!

File: package.json (hunk 1 above)
Comment (kentmclean) [reply package.json]: I also have a change set queued up that removes support for rx4 altogether. Welcome thoughts on that as well.

File: .travis.yml [hunk 3]
```diff
@@ -1,3 +1,3 @@
 language: node_js
 node_js:
-  - "7"
+  - "8"
```
Comment (kentmclean) [inline .travis.yml]: necessary for builds to succeed. Yarn doesn't support node@7 evidently

File: package.json (hunk 1 above)
Comment (geoctrl) [reply package.json]: why is `peerDependencies` not a good option?

File: .travis.yml (hunk 3 above)
Comment (geoctrl) [reply .travis.yml]: 10? 🤷‍♂️

File: package.json (hunk 1 above)
Comment (TheMcMurder) [reply package.json]: I'm curious about that also, does `peerDependencies` not work?

Comment (kentmclean) [reply package.json]: yes, with a caveat that those dependencies would need to be included in devDependencies also. Happy to experiment with that a bit

**Pull request merged**